scripts/generators/api_dump_generator.py

```diff
@@ -0,0 +1,1320 @@
+        'stencilAttachmentFormat': '!ApiDumpInstance::current().getIsGPLPreRasterOrFragmentShader()',
+    },
+    'VkIndirectExecutionSetInfoEXT':{
+        'pPipelineInfo': 'ApiDumpInstance::current().getIndirectExecutionSetInfoType() == VK_INDIRECT_EXECUTION_SET_INFO_TYPE_PIPELINES_EXT',
+        'pShaderInfo': 'ApiDumpInstance::current().getIndirectExecutionSetInfoType() == VK_INDIRECT_EXECUTION_SET_INFO_TYPE_SHADER_OBJECTS_EXT',
+    },
+    'VkIndirectCommandsTokenDataEXT':{
+        'pPushConstant': 'ApiDumpInstance::current().getIndirectCommandsLayoutToken() == VK_INDIRECT_COMMANDS_TOKEN_TYPE_PUSH_CONSTANT_EXT || ApiDumpInstance::current().getIndirectCommandsLayoutToken() == VK_INDIRECT_COMMANDS_TOKEN_TYPE_SEQUENCE_INDEX_EXT',
+        'pVertexBuffer': 'ApiDumpInstance::current().getIndirectCommandsLayoutToken() == VK_INDIRECT_COMMANDS_TOKEN_TYPE_VERTEX_BUFFER_EXT',
+        'pIndexBuffer': 'ApiDumpInstance::current().getIndirectCommandsLayoutToken() == VK_INDIRECT_COMMANDS_TOKEN_TYPE_INDEX_BUFFER_EXT',
+        'pExecutionSet': 'ApiDumpInstance::current().getIndirectCommandsLayoutToken() == VK_INDIRECT_COMMANDS_TOKEN_TYPE_EXECUTION_SET_EXT',
+    },
+}
+
+# These types are defined in both video.xml and vk.xml. Because duplicate functions aren't allowed,
+# we have to prevent these from generating twice. This is done by removing the types from the non-video
+# outputs
+DUPLICATE_TYPES_IN_VIDEO_HEADER = ['uint32_t', 'uint16_t', 'uint8_t', 'int32_t', 'int16_t', 'int8_t']
+
+# Short list of all of the functions that are 'global', meaning they can be queried from vkGetInstanceProcAddr(NULL, "<func_name>")
+GLOBAL_FUNCTION_NAMES = ['vkEnumerateInstanceLayerProperties', 'vkEnumerateInstanceExtensionProperties', 'vkEnumerateInstanceVersion', 'vkCreateInstance']
+
+BLOCKING_API_CALLS = [
+    'vkWaitForFences', 'vkWaitSemaphores', 'vkQueuePresentKHR', 'vkDeviceWaitIdle',
+    'vkQueueWaitIdle', 'vkAcquireNextImageKHR', 'vkGetQueryPoolResults', 'vkWaitSemaphoresKHR'
+]
+
+EXCLUDED_INCLUDE_LIST = [
+    'vk_platform',
+    'stdint.h',
+]
+
+FUNCTION_IMPLEMENTATION_IGNORE_LIST = ['vkGetDeviceProcAddr', 'vkGetInstanceProcAddr', 'vkEnumerateInstanceVersion']
+
+class PlatformGuardHelper():
+    """Used to elide platform guards together, so redundant #endif then #ifdefs are removed
+    Note - be sure to call add_guard(None) when done to add a trailing #endif if needed
+    """
+    def __init__(self):
+        self.current_guard = None
+
+    def add_guard(self, generator, guard):
+        if self.current_guard != guard and self.current_guard is not None:
+            generator.write(f'#endif  // {self.current_guard}')
+        if self.current_guard != guard and guard is not None:
+            generator.write(f'#if defined({guard})')
+        self.current_guard = guard
+
+# Returns the command's parameters declaration. ex) uint32_t foo, float bar, char* baz
+def commandParameterDeclarationText(command):
```

Comment:
nit: (and below)

```suggestion
def commandParameterDeclarationText(command: Command):
```

also maybe match function naming_style_here?